assets/js/components/OverlayNotification/OverlayNotificationsRenderer.js

```diff
@@ -29,19 +29,37 @@ import AudienceSegmentationIntroductoryOverlayNotification from '../../modules/a
 import AnalyticsAndAdSenseAccountsDetectedAsLinkedOverlayNotification from './AnalyticsAndAdSenseAccountsDetectedAsLinkedOverlayNotification';
 import LinkAnalyticsAndAdSenseAccountsOverlayNotification from './LinkAnalyticsAndAdSenseAccountsOverlayNotification';
 import { PublicationApprovedOverlayNotification } from '../../modules/reader-revenue-manager/components/dashboard';
+import { useSelect } from 'googlesitekit-data';
+import { MODULES_READER_REVENUE_MANAGER } from '../../modules/reader-revenue-manager/datastore/constants';
+import { CORE_MODULES } from '../../googlesitekit/modules/datastore/constants';
 
 export default function OverlayNotificationsRenderer() {
 	const audienceSegmentationEnabled = useFeature( 'audienceSegmentation' );
 	const readerRevenueManagerEnabled = useFeature( 'rrmModule' );
+	const analyticsConnected = useSelect( ( select ) =>
+		select( CORE_MODULES ).isModuleConnected( 'modules/analytics-4' )
+	);
+	const rrmConnected = useSelect( ( select ) =>
+		select( CORE_MODULES ).isModuleConnected(
+			MODULES_READER_REVENUE_MANAGER
+		)
+	);
+	const adsenseConnected = useSelect( ( select ) =>
+		select( CORE_MODULES ).isModuleConnected( 'modules/adsense' )
+	);
 
 	return (
 		<Fragment>
-			<LinkAnalyticsAndAdSenseAccountsOverlayNotification />
-			<AnalyticsAndAdSenseAccountsDetectedAsLinkedOverlayNotification />
-			{ audienceSegmentationEnabled && (
+			{ analyticsConnected && adsenseConnected && (
+				<Fragment>
+					<LinkAnalyticsAndAdSenseAccountsOverlayNotification />
+					<AnalyticsAndAdSenseAccountsDetectedAsLinkedOverlayNotification />
+				</Fragment>
+			) }
+			{ analyticsConnected && audienceSegmentationEnabled && (
 				<AudienceSegmentationIntroductoryOverlayNotification />
 			) }
-			{ readerRevenueManagerEnabled && (
+			{ rrmConnected && readerRevenueManagerEnabled && (
 				<PublicationApprovedOverlayNotification />
```

Comment:
As a convention, we generally check for the feature flag first before anything else although in this case it doesn't make a difference. In other situations, it could if the condition involved computation or something more expensive.

Instead of adding the `isConnected` checks here, let's wrap the components with `isActive` (using the appropriate arguments to ensure the behavior is the same). The result is the same but makes for a cleaner approach, but also makes sense so that if the component was used in multiple places, the same condition wouldn't need to be repeated as it would always be relevant.

Reply:
Correct, I meant `whenActive`! my mistake 😄